models/items.js

```diff
@@ -14,24 +12,26 @@ const addTagsToItem = async (itemData) => {
   try {
     const { itemId, itemType, tagPayload } = itemData;
     const batch = firestore.batch();
+    let isNewtag = false;
     for (const tag of tagPayload) {
-      const tagData = await tagModel.doc(tag.tagId).get();
-      const levelData = await levelModel.doc(tag.levelId).get();
+      const itemData = await itemTagsModel
+        .where("itemId", "==", itemId)
+        .where("tagId", "==", tag.tagId)
+        .where("levelId", "==", tag.levelId)
+        .get();
+      if (!itemData.empty) continue;
+      isNewtag = true;
       const itemTag = {
         itemId,
         itemType: itemType.toUpperCase(),
         tagId: tag.tagId,
-        tagName: tagData.data().name,
-        tagType: tagData.data().type,
         levelId: tag.levelId,
-        levelName: levelData.data().name,
-        levelValue: levelData.data().value,
```

Comment:
have we missed fetching these details on the get API call for getting the item based on filter, should we not fetch all the details of tag and levels and send it when we are fetching the item

Reply:
If we remove these values, we can only filter based on tags and levels(tagId and levelId specifically). Other types of filtering wont work as of now